src/App.tsx

```diff
@@ -1,14 +1,127 @@
 import { AuthContext } from './components/Auth/AuthContext';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoInfo } from './components/TodoInfo/TodoInfo';
+import { TodoList } from './components/TodoList/TodoList';
+import { FilterTypes } from './types/Enums';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterType, setFilterType] = useState('All');
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const handleButtonClickAll = () => {
+    setFilterType(FilterTypes.All);
+  };
+
+  const handleButtonClickActive = () => {
+    setFilterType(FilterTypes.ACTIVE);
+  };
+
+  const handleButtonClickCompleted = () => {
+    setFilterType(FilterTypes.COMPLETED);
+  };
+
+  const onSubmit = useCallback((event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      setErrorMessage('Title can not be empty');
+
+      return;
+    }
+
+    if (user) {
+      setTempTodo({
+        id: 0,
+        title: todoTitle,
+        completed: false,
+        userId: user.id,
+      });
+
+      setIsLoading(true);
+      addTodo(todoTitle, user.id)
+        .then(response => {
+          setTodos(prev => [...prev, {
+            id: response.id,
+            title: response.title,
+            completed: response.completed,
+            userId: response.userId,
+          }]);
+        })
+        .catch(() => setErrorMessage('Unable to add a todo'))
+        .finally(() => {
+          setIsLoading(false);
+          setTempTodo(null);
+          setTodoTitle('');
+        });
+    }
+  }, [todoTitle, user]);
+
+  const onDeleteTodo = useCallback((id: number) => {
+    setIsLoading(true);
+    deleteTodo(id)
+      .then(() => (
+        setTodos(currentTodos => currentTodos.filter(todo => todo.id !== id))
+      ))
+      .catch(() => {
+        setErrorMessage('Impossible to delete todo');
+      });
+    setIsLoading(false);
+  }, []);
+
+  const onClickClearCompleted = () => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        onDeleteTodo(todo.id);
+      }
+    });
+  };
+
+  const visibleTodos = useMemo(() => {
+    switch (filterType) {
+      case FilterTypes.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+
+      case FilterTypes.COMPLETED:
+        return todos.filter(todo => todo.completed);
+
+      default:
```

Comment:
Use all cases, it will improve productivity
```suggestion

      case FilterTypes.ALL:
      default:
```